packages/ember-metal/lib/path_cache.js

```diff
@@ -4,10 +4,10 @@ var IS_GLOBAL      = /^([A-Z$]|([0-9][A-Z$]))/;
 var IS_GLOBAL_PATH = /^([A-Z$]|([0-9][A-Z$])).*[\.]/;
 var HAS_THIS       = 'this.';
 
-var isGlobalCache       = new Cache(1000, function(key) { return IS_GLOBAL.test(key);          });
-var isGlobalPathCache   = new Cache(1000, function(key) { return IS_GLOBAL_PATH.test(key);     });
-var hasThisCache        = new Cache(1000, function(key) { return key.lastIndexOf(HAS_THIS, 0) === 0; });
-var firstDotIndexCache  = new Cache(1000, function(key) { return key.indexOf('.');             });
+var isGlobalCache       = new Cache(1000, (key) => { IS_GLOBAL.test(key); });
+var isGlobalPathCache   = new Cache(1000, (key) => { IS_GLOBAL_PATH.test(key); });
+var hasThisCache        = new Cache(1000, (key) => { return key.lastIndexOf(HAS_THIS, 0) === 0; });
+var firstDotIndexCache  = new Cache(1000, (key) => { key.indexOf('.'); });
```

Comment:
> The version with no brackets will be a little confusing for some, unless they know ruby or coffeescript

...or ES6?

Sure thing, but either way it needs to return the value in this particular case, so three options:

```js
// Multiline arrow function, explicit return
var isGlobalPathCache = new Cache(1000, key => {
  return IS_GLOBAL_PATH.test(key);
});

// Single line arrow function, implicit return (no curly braces)
var isGlobalPathCache = new Cache(1000, key => IS_GLOBAL_PATH.test(key));

// Oldschool anonymous function, explicit return
var isGlobalPathCache = new Cache(1000, function(key) {
  return IS_GLOBAL_PATH.test(key);
});
```